test/e2e/nsx_ipaddressallocation_test.go

```diff
@@ -0,0 +1,136 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+)
+
+const (
+	externalCIDR   = "192.168.0.0/27"
+	privateCIDR    = "172.26.0.0/27"
+	privateTGWCIDR = "10.246.0.0/27"
+)
+
+func TestIPAddressAllocation(t *testing.T) {
+	t.Run("testIPAddressAllocationExternal", func(t *testing.T) {
+		testIPAddressAllocation(t, "./manifest/testIPAddressAllocation/ipaddressallocation_external.yaml", externalCIDR)
+	})
+	t.Run("testIPAddressAllocationPrivate", func(t *testing.T) {
+		testIPAddressAllocation(t, "./manifest/testIPAddressAllocation/ipaddressallocation_private.yaml", privateCIDR)
+	})
+	t.Run("testIPAddressAllocationPrivateTGW", func(t *testing.T) {
+		testIPAddressAllocation(t, "./manifest/testIPAddressAllocation/ipaddressallocation_privatetgw.yaml", privateTGWCIDR)
+	})
+}
+
+func testIPAddressAllocation(t *testing.T, yamlPath string, expectedCIDR string) {
+	deadlineCtx, deadlineCancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer deadlineCancel()
+
+	ns := "test-ipaddress-allocation"
+	var err error
+
+	setupTest(t, ns)
+	defer teardownTest(t, ns, defaultTimeout)
+
+	// Parse YAML to get CR's name
+	ipAllocPath, _ := filepath.Abs(yamlPath)
+	require.NoError(t, applyYAML(ipAllocPath, ns))
+	defer func(filename string, ns string) {
+		_ = deleteYAML(filename, ns)
+	}(ipAllocPath, ns)
```

Comment:
Do we still need this as we always delete the IPAddressAllocation in this test as line 61-62?

Reply:
fixed